spec/rspec/support/object_inspector_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+require 'rspec/support/object_inspector'
+require 'rspec/matchers/fail_matchers'
+
+module RSpec
+  module Support
+    describe ObjectInspector, ".inspect" do
+      include RSpec::Matchers::FailMatchers
+
+      context 'with Time objects' do
+        let(:time) { Time.utc(1969, 12, 31, 19, 01, 40, 101) }
+        let(:formatted_time) { ObjectInspector.inspect(time) }
+
+        it 'produces an extended output' do
+          expected_output = "1969-12-31 19:01:40.000101"
+          expect(formatted_time).to include(expected_output)
+        end
+      end
+
+      context 'with DateTime objects' do
+        def with_date_loaded
+          in_sub_process_if_possible do
+            require 'date'
+            yield
+          end
+        end
+
+        let(:date) { DateTime.new(2000, 1, 1, 1, 1, Rational(1, 10)) }
+        let(:formatted_date) { ObjectInspector.inspect(date) }
```

Comment:
I think it's a little confusing to call these `date` and `formatted_date` given you are dealing with a `DateTime` rather than a `Date` object.